highfidelity/hifi
# Material reflection

Using the new graphics scripting interface, you can query a shape entity, model entity, model overlay, or avatar for its materials and material names (if it has any).

This allows you to easily copy the materials from things into new material entities to apply them to other things.

Test plan:
- run [this](https://gist.githubusercontent.com/SamGondelman/f095df9fdc0a93272343374fc45850f6/raw/f66125e6fcb0605571bdd0ce212765026dbd9c6a/MaterialPicker.js) in an empty domain where you have edit rights.
- you should see two boxes and three models in front of you
- click the two boxes.  The middle mesh should change color to match the cube you click.
- when you click a box, an empty line should print in the log.
- click on various submeshes of the top right model (try: bow, shirt, arm).  The middle model should change appearance each time (the textures will be jumbled but that's fine).
- when you click on the top right model, this should print in the logs: `StingrayPBS3,StingrayPBS2,StingrayPBS1,StingrayPBS3,StingrayPBS1`
- click the bottom right model.  The middle model should return to normal.
- this should be printed in the logs: `StingrayPBS1`
- go into third person and click in the middle of your avatar.  If you hit it (avatar raypicks collide against the T-pose, not the animated mesh), the middle model should change appearance (what it looks like will depend on what avatar you're wearing and which submesh you select)
